### ISSUE TITLE
The parameters BOX(P, R, mAP50, mAP50-95) are always 0 when the batch value is high.

### ISSUE THREAD

**Issue by sascha93rus** [ultralytics/ultralytics#16302]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

**I am using a laptop with a GTX1650 graphics card with 4 GB of video memory. I train the YOLOv10m model, but the choice of model does not affect the result (I tried Yolov8n). Amp is disabled because it does not work on this video card. With batch = 4, during training, the BOX parameters (P, R, mAP50, mAP50-95) are always 0. At the same time, video memory consumption = 1.45 GB. Because of this, determining the best model does not work. If you reduce the batch value to 3, then everything works (video memory consumption = 1.2GB). With automatic batch detection (batch = -1), training takes place with an increased batch (batch = 5). This problem does not allow efficient use of the GPU.**

Batch = 4:
Ultralytics YOLOv8.2.94 🚀 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
engine\trainer: task=detect, mode=train, model=yolov10m.pt, data=./yolo_data_symbol/data.yaml, epochs=200, time=None, patience=100, batch=4, imgsz=288, save=True, save_period=-1, cache=False, device=cuda, workers=8, project=None, name=symbolDetectV10m3, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=False, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\symbolDetectV10m3  
Overriding model.yaml nc=80 with nc=22

                   from  n    params  module                                       arguments
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]
  5                  -1  1     78720  ultralytics.nn.modules.block.SCDown          [192, 384, 3, 2]
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]
  7                  -1  1    228672  ultralytics.nn.modules.block.SCDown          [384, 576, 3, 2]
  8                  -1  2   1689984  ultralytics.nn.modules.block.C2fCIB          [576, 576, 2, True]
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]
 10                  -1  1   1253088  ultralytics.nn.modules.block.PSA             [576, 576]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  2   1993728  ultralytics.nn.modules.block.C2f             [960, 384, 2]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  2    517632  ultralytics.nn.modules.block.C2f             [576, 192, 2]
 17                  -1  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  2    831744  ultralytics.nn.modules.block.C2fCIB          [576, 384, 2, True]
 20                  -1  1    152448  ultralytics.nn.modules.block.SCDown          [384, 384, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  2   1911168  ultralytics.nn.modules.block.C2fCIB          [960, 576, 2, True]
 23        [16, 19, 22]  1   2306452  ultralytics.nn.modules.head.v10Detect        [22, [192, 384, 576]]
YOLOv10m summary: 498 layers, 16,509,604 parameters, 16,509,588 gradients, 64.1 GFLOPs

Transferred 787/799 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
train: Scanning D:\py\yolo_data_symbol\train\labels.cache... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:00<?, ?it/s]
val: Scanning D:\py\yolo_data_symbol\valid\labels.cache... 140 images, 0 backgrounds, 0 corrupt: 100%|██████████| 140/140 [00:00<?, ?it/s]
Plotting labels to runs\detect\symbolDetectV10m3\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: AdamW(lr=0.000385, momentum=0.9) with parameter groups 129 weight(decay=0.0), 142 weight(decay=0.0005), 141 bias(decay=0.0)
Image sizes 288 train, 288 val
Using 8 dataloader workers
Logging results to runs\detect\symbolDetectV10m3
Starting training for 200 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/200      1.32G      1.789      4.298      1.802         53        288: 100%|██████████| 200/200 [00:51<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:04<00:00,  4.16it/s]
                   all        140       1166          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/200      1.45G      1.409      1.322      1.692         47        288: 100%|██████████| 200/200 [00:52<00:00,  3.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:04<00:00,  4.35it/s]
                   all        140       1166          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/200      1.44G      1.357      1.026       1.69         17        288: 100%|██████████| 200/200 [00:55<00:00,  3.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:03<00:00,  4.82it/s]
                   all        140       1166          0          0          0          0
_______________________________________________________________________________________________________________________________________

batch = 3:
Ultralytics YOLOv8.2.94 🚀 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
engine\trainer: task=detect, mode=train, model=yolov10m.pt, data=./yolo_data_symbol/data.yaml, epochs=200, time=None, patience=100, batch=3, imgsz=288, save=True, save_period=-1, cache=False, device=cuda, workers=8, project=None, name=symbolDetectV10m2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=False, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\symbolDetectV10m2  
Overriding model.yaml nc=80 with nc=22

                   from  n    params  module                                       arguments
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]
  5                  -1  1     78720  ultralytics.nn.modules.block.SCDown          [192, 384, 3, 2]
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]
  7                  -1  1    228672  ultralytics.nn.modules.block.SCDown          [384, 576, 3, 2]
  8                  -1  2   1689984  ultralytics.nn.modules.block.C2fCIB          [576, 576, 2, True]
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]
 10                  -1  1   1253088  ultralytics.nn.modules.block.PSA             [576, 576]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  2   1993728  ultralytics.nn.modules.block.C2f             [960, 384, 2]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  2    517632  ultralytics.nn.modules.block.C2f             [576, 192, 2]
 17                  -1  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  2    831744  ultralytics.nn.modules.block.C2fCIB          [576, 384, 2, True]
 20                  -1  1    152448  ultralytics.nn.modules.block.SCDown          [384, 384, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  2   1911168  ultralytics.nn.modules.block.C2fCIB          [960, 576, 2, True]
 23        [16, 19, 22]  1   2306452  ultralytics.nn.modules.head.v10Detect        [22, [192, 384, 576]]
YOLOv10m summary: 498 layers, 16,509,604 parameters, 16,509,588 gradients, 64.1 GFLOPs

Transferred 787/799 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
train: Scanning D:\py\yolo_data_symbol\train\labels.cache... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:00<?, ?it/s]
val: Scanning D:\py\yolo_data_symbol\valid\labels.cache... 140 images, 0 backgrounds, 0 corrupt: 100%|██████████| 140/140 [00:00<?, ?it/s]
Plotting labels to runs\detect\symbolDetectV10m2\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: AdamW(lr=0.000385, momentum=0.9) with parameter groups 129 weight(decay=0.0), 142 weight(decay=0.0004921875), 141 bias(decay=0.0)
Image sizes 288 train, 288 val
Using 8 dataloader workers
Logging results to runs\detect\symbolDetectV10m2
Starting training for 200 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/200      1.08G      1.765      4.128      1.807         48        288: 100%|██████████| 266/266 [01:06<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 24/24 [00:04<00:00,  4.83it/s]
                   all        140       1166      0.635     0.0111     0.0209     0.0199

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/200      1.21G      1.448      1.309      1.702         57        288:  73%|███████▎  | 195/266 [00:46<00:18,  3.82it/s]

### Environment

Ultralytics YOLOv8.2.94 🚀 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
Setup complete ✅ (12 CPUs, 23.4 GB RAM, 63.1/292.9 GB disk)

### Minimal Reproducible Example

import torch
from ultralytics import YOLO

if __name__ == '__main__':

    device = 'cuda' if torch.cuda.is_available() else 'cpu'

    data_path = './yolo_data_symbol/data.yaml'  

    model = YOLO("yolov10m.pt", task='detect') 

    epochs_size = 200  
    batch = 4  
    imgsize = 288  

    model.train(name='symbolDetectV10m', data=data_path, epochs=epochs_size, batch=batch, imgsz=imgsize, device=device, amp=False, half=False)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sascha93rus, thank you for bringing up this issue with Ultralytics 🚀! This is an automated response, but don't worry, an Ultralytics engineer will take a look at your question soon.

We recommend exploring the [Docs](https://docs.ultralytics.com) for guidance, where many of the Python and CLI use cases are detailed. 🔍

If this is a 🐛 bug report, thank you for providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This is very helpful for our debugging process.

Please ensure you are using the latest `ultralytics` package with all dependencies satisfied. You can upgrade using:

```bash
pip install -U ultralytics
```

If you have further details or encounters, feel free to share them for more insightful assistance.

Join our vibrant community for real-time communication on [Discord](https://ultralytics.com/discord) 🎧, or engage in discussions on [Discourse](https://community.ultralytics.com) and our [Subreddit](https://reddit.com/r/ultralytics).

## Verified Environments

YOLOv8 can be run in several verified environments. Options include:

- **Notebooks**: Free GPU access on <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud**: Google Cloud or AWS Deep Learning VMs
- **Docker**: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, verifying the integrity of YOLOv8 across platforms and tasks.

Thank you for your detailed report! 😊

**Comment by Y-T-G** [comment]
Does inference work for you with `half=True`? Can you try it with any of the pre-trained models and also see if the results are correct or random?

```python
from ultralytics import YOLO

model = YOLO("yolov8n.pt")
results = model("ultralytics/assets/bus.jpg", half=True, show=True)
```

**Comment by sascha93rus** [comment]
I tried to train the model with half=true and amp=false, but it didn't help:

```
Ultralytics YOLOv8.2.94 🚀 Python-3.12.3 torch-2.4.1+cu124 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
engine\trainer: task=detect, mode=train, model=yolov10m.pt, data=./yolo_data_symbol/data.yaml, epochs=200, time=None, patience=100, batch=4, imgsz=288, save=True, save_period=-1, cache=False, device=cuda, workers=8, project=None, name=symbolDetectV10m4, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=False, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=True, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\detect\symbolDetectV10m4
Overriding model.yaml nc=80 with nc=22

                   from  n    params  module                                       arguments
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]
  5                  -1  1     78720  ultralytics.nn.modules.block.SCDown          [192, 384, 3, 2]
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]
  7                  -1  1    228672  ultralytics.nn.modules.block.SCDown          [384, 576, 3, 2]
  8                  -1  2   1689984  ultralytics.nn.modules.block.C2fCIB          [576, 576, 2, True]
  9                  -1  1    831168  ultralytics.nn.modules.block.SPPF            [576, 576, 5]
 10                  -1  1   1253088  ultralytics.nn.modules.block.PSA             [576, 576]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  2   1993728  ultralytics.nn.modules.block.C2f             [960, 384, 2]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  2    517632  ultralytics.nn.modules.block.C2f             [576, 192, 2]
 17                  -1  1    332160  ultralytics.nn.modules.conv.Conv             [192, 192, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  2    831744  ultralytics.nn.modules.block.C2fCIB          [576, 384, 2, True]
 20                  -1  1    152448  ultralytics.nn.modules.block.SCDown          [384, 384, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  2   1911168  ultralytics.nn.modules.block.C2fCIB          [960, 576, 2, True]
 23        [16, 19, 22]  1   2306452  ultralytics.nn.modules.head.v10Detect        [22, [192, 384, 576]]
YOLOv10m summary: 498 layers, 16,509,604 parameters, 16,509,588 gradients, 64.1 GFLOPs

Transferred 787/799 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
train: Scanning D:\py\yolo_data_symbol\train\labels.cache... 798 images, 0 backgrounds, 0 corrupt: 100%|██████████| 798/798 [00:00<?, ?it/s]
val: Scanning D:\py\yolo_data_symbol\valid\labels.cache... 140 images, 0 backgrounds, 0 corrupt: 100%|██████████| 140/140 [00:00<?, ?it/s]
Plotting labels to runs\detect\symbolDetectV10m4\labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: AdamW(lr=0.000385, momentum=0.9) with parameter groups 129 weight(decay=0.0), 142 weight(decay=0.0005), 141 bias(decay=0.0)
Image sizes 288 train, 288 val
Using 8 dataloader workers
Logging results to runs\detect\symbolDetectV10m4
Starting training for 200 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      1/200      1.32G      1.789      4.298      1.802         53        288: 100%|██████████| 200/200 [00:48<00:00,  4.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:04<00:00,  3.99it/s]
                   all        140       1166          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      2/200      1.45G      1.409      1.322      1.692         47        288: 100%|██████████| 200/200 [00:44<00:00,  4.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:03<00:00,  5.11it/s]
                   all        140       1166          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      3/200      1.44G      1.357      1.026       1.69         17        288: 100%|██████████| 200/200 [00:43<00:00,  4.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:03<00:00,  5.09it/s]
                   all        140       1166          0          0          0          0

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      4/200      1.45G      1.316     0.9029      1.673         21        288: 100%|██████████| 200/200 [00:43<00:00,  4.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 18/18 [00:03<00:00,  5.21it/s]
                   all        140       1166          0          0          0          0
```

I tried this code and this is what happened:
![image](https://github.com/user-attachments/assets/fc39652f-37f0-489a-ac31-68505beb26ef)

**Comment by Y-T-G** [comment]
@sascha93rus 

Can you run inference with the following?

```
from ultralytics import YOLO
import cv2

img = cv2.imread("ultralytics/assets/bus.jpg")
batch = [img] * 4

model = YOLO("yolov8n.pt")
results = model(batch, half=True)
```

**Comment by sascha93rus** [comment]
![image](https://github.com/user-attachments/assets/a8d18a3d-5d2e-43b8-9aa1-00eb6f906051)

**Comment by Y-T-G** [comment]
@sascha93rus 

Try with `batch=8`:

```
from ultralytics import YOLO
import cv2

img = cv2.imread("ultralytics/assets/bus.jpg")
batch = [img] * 8

model = YOLO("yolov8n.pt")
results = model(batch, half=True)
```

**Comment by sascha93rus** [comment]
I ran some more tests and tried to train a simpler YOLOv10n model. Now these indicators (mAP50, etc.) are equal to 0 with the value batch = 7. When the batch value is 6, these indicators work.

As for run inference:
![image](https://github.com/user-attachments/assets/427eb7ec-2bd6-42b7-8a4c-ca062bc367d4)

But at the same time, for example, batch = 32 works:
![image](https://github.com/user-attachments/assets/3e8be591-1a82-4f60-9971-86cfb882bba6)

**Comment by Y-T-G** [comment]
@sascha93rus Hmm. Interesting. But the results of `batch=32` are wrong too because there are only 4 persons in the image and 1 bus.

Can you also run this:
```
from ultralytics import YOLO
from ultralytics.utils.torch_utils import autocast
import cv2
import torch

img = cv2.imread("ultralytics/assets/bus.jpg")

model = YOLO("yolov8n.pt").to("cuda:0")

def amp_checks(m, im, imgsz=640, bs=8):
    device = next(model.parameters()).device
    batch = [im] * bs
    a = m(batch, imgsz=imgsz, device=device, verbose=False)[0].boxes.data  # FP32 inference
    with autocast(enabled=True):
        b = m(batch, imgsz=imgsz, device=device, verbose=False)[0].boxes.data  # AMP inference
    m.predictor.model.fp16 = True
    m.predictor.model.half()
    c = m(batch, imgsz=imgsz, device=device, verbose=False, half=True)[0].boxes.data  # Half inference
    del m
    amp = torch.allclose(a, b.float(), atol=0.5)
    half = torch.allclose(a, c.float(), atol=1.0)
    all = a.shape == b.shape == c.shape and amp and half
    return amp, half, all

print("Test with imgsz=640, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img)))
model.predictor = None

print("Test with imgsz=320, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=320, bs=8)))
model.predictor = None

print("Test with imgsz=160, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=160, bs=8)))
model.predictor = None

print("Test with imgsz=96, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=96, bs=8)))
```

**Comment by sascha93rus** [comment]
@Y-T-G I am getting this error:
![image](https://github.com/user-attachments/assets/8c5997bb-b520-4e6f-9447-ab2bc565f006)

**Comment by Y-T-G** [comment]
Hmm. Try this:

```python
from ultralytics import YOLO
from ultralytics.utils.torch_utils import autocast
import cv2
import torch

img = cv2.imread("ultralytics/assets/bus.jpg")

model = YOLO("yolov8n.pt").to("cuda:0")

def amp_checks(m, im, imgsz=640, bs=8):
    try:
        device = next(model.parameters()).device
        batch = [im] * bs
        a = m(batch, imgsz=imgsz, device=device, verbose=False)[0].boxes.data  # FP32 inference
        with autocast(enabled=True):
            b = m(batch, imgsz=imgsz, device=device, verbose=False)[0].boxes.data  # AMP inference
        m.predictor.model.fp16 = True
        m.predictor.model.half()
        c = m(batch, imgsz=imgsz, device=device, verbose=False, half=True)[0].boxes.data  # Half inference
        del m
        amp = torch.allclose(a, b.float(), atol=0.5)
        half = torch.allclose(a, c.float(), atol=1.0)
        all = a.shape == b.shape == c.shape and amp and half
        return amp, half, all
    except:
        print(f"Failed imgsz={imgsz}, bs={bs}")
        return False, False, False

print("Test with imgsz=640, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img)))
model.predictor = None

print("Test with imgsz=320, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=320, bs=8)))
model.predictor = None

print("Test with imgsz=160, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=160, bs=8)))
model.predictor = None

print("Test with imgsz=96, bs=8: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=96, bs=8)))
model.predictor = None

print("Test with imgsz=640, bs=1: Autocast:{}, Half:{}, Overall:{}".format(*amp_checks(model, img, imgsz=640, bs=1)))
```

**Comment by Y-T-G** [comment]
@sascha93rus

**Comment by sascha93rus** [comment]
@Y-T-G 
![image](https://github.com/user-attachments/assets/f75acdd7-1846-4207-b4ef-168978249744)

**Comment by Y-T-G** [comment]
@sascha93rus Thanks for the tests. I think I understand the issue now.

You can use [this](https://github.com/ultralytics/ultralytics/issues/15248#issuecomment-2283093429) to fix the validation issue for now. The PR I opened should resolve this issue too.

**Comment by sascha93rus** [comment]
@Y-T-G Thank you! I'll wait for the release.

**Comment by glenn-jocher** [comment]
Great news 😃! Your original issue may now be resolved ✅ in PR #16306. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
